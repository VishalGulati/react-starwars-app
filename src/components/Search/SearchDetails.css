.search-details {
  text-transform: capitalize;
  font-size: 12px;
}