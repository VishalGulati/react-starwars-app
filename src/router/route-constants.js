export const loginRoute = 'login';
export const searchRoute = 'search';
export const searchDetailsRoute = 'searchDetails';
