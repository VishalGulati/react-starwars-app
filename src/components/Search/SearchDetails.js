import React, { Component } from 'react';
import {
    Table
} from 'react-bootstrap';
import './SearchDetails.css';

export default class SearchDetails extends Component {
    render() {
        let searchItemDetails = [];
        if (this.props.location.params) {
            const params = this.props.location.params;
            searchItemDetails = Object.keys(
                params
            ).map((propName, key) => {
                return <tr key={key}>
                    <td>{propName}</td>
                    <td>{params[propName]}</td>
                </tr>
            })
        }

        return (
            <div >
                <h1>Search Details</h1>
                <Table striped bordered condensed hover className="search-details">
                    <tbody>
                        {searchItemDetails}
                    </tbody>
                </Table>
            </div>
        );
    }
}
