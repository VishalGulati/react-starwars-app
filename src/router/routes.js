import React from 'react';
import { Route } from 'react-router-dom';
import { loginRoute, searchRoute, searchDetailsRoute } from './route-constants';
import Login from '../components/Login/Login';
import Search from '../components/Search/Search';
import SearchDetails from '../components/Search/SearchDetails';

// route config
const routes = [
  // {
  //   path: '/',
  //   component: Login,
  //   pathMatch: 'full'
  // },
  {
    path: `/`,
    component: Login
  },
  {
    path: `/${searchRoute}`,
    component: Search
  },
  {
    path: `/${searchDetailsRoute}`,
    component: SearchDetails
  }
];

// wrap <Route> and use this everywhere instead, then when
// sub routes are added to any route it'll work
const RouteWithSubRoutes = route => (
  <Route
    path={route.path}
    render={props => (
      // pass the sub-routes down to keep nesting
      <route.component {...props} routes={route.routes} />
    )}
  />
);

const CustomRoutes = () => (
  <main>
    {routes.map((route, i) => <RouteWithSubRoutes key={i} {...route} />)}
  </main>
);

export default CustomRoutes;
