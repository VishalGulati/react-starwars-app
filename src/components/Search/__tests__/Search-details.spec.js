import React from 'react';
import renderer from 'react-test-renderer';
import SearchDetails from '../SearchDetails';
import { searchDetailsRoute } from '../../router/route-constants'

describe('Search', () => {

    function props() {
        return {
            location: {
                pathname: `/${searchDetailsRoute}`,
                params: {
                    "name": "Tatooine",
                    "rotation_period": "23",
                    "orbital_period": "304",
                    "diameter": "10465",
                    "climate": "arid",
                    "gravity": "1 standard",
                    "terrain": "desert",
                    "surface_water": "1",
                    "population": "200000",
                }
            }
        }
    }

    it('should render Search Details properly', () => {
        const component = renderer.create(<SearchDetails {...props() } />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
