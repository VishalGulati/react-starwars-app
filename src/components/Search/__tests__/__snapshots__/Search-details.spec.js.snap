// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render Search Details properly 1`] = `
<table
  className="search-details table table-striped table-bordered table-condensed table-hover"
>
  <tbody>
    <tr>
      <td>
        name
      </td>
      <td>
        Tatooine
      </td>
    </tr>
    <tr>
      <td>
        rotation_period
      </td>
      <td>
        23
      </td>
    </tr>
    <tr>
      <td>
        orbital_period
      </td>
      <td>
        304
      </td>
    </tr>
    <tr>
      <td>
        diameter
      </td>
      <td>
        10465
      </td>
    </tr>
    <tr>
      <td>
        climate
      </td>
      <td>
        arid
      </td>
    </tr>
    <tr>
      <td>
        gravity
      </td>
      <td>
        1 standard
      </td>
    </tr>
    <tr>
      <td>
        terrain
      </td>
      <td>
        desert
      </td>
    </tr>
    <tr>
      <td>
        surface_water
      </td>
      <td>
        1
      </td>
    </tr>
    <tr>
      <td>
        population
      </td>
      <td>
        200000
      </td>
    </tr>
  </tbody>
</table>
`;
